### ISSUE TITLE
Remove the GraphQL.ExecutionError prefix in error message

### ISSUE THREAD

**Issue by mariusbancila** [graphql-dotnet/graphql-dotnet#3327]
I have the following (simplified) pattern in my code:
```
Resolver = new FuncFieldResolver<object>(ctx =>
{
   try
   {
      // ....
   }
   catch(Exception ex)
   {
      ctx.Errors.Add(new ExecutionError(ex.Message));
   }

   return null;
}),
```

When some exceptions occur, I add a new `ExecutionError` to the context's `Errors` collection. But the, the error messages look like this:
```
"errors": [
    {
      "message": "GraphQL.ExecutionError: Invalid pagination arguments."
    }
  ],
```
How can I get rid of this `GraphQL.ExecutionError:` prefix? I don't see anything to control this in the `ErrorInfoProviderOptions` class.

**Comment by Shane32** [comment]
Which version of GraphQL.NET?

**Comment by mariusbancila** [comment]
The version I am using is `5.3.3`.

**Comment by Shane32** [comment]
Normally that prefix is only added when an unhandled exception occurs, which can be changed by setting the unhandled execution error delegate.  See [Processing Errors](https://graphql-dotnet.github.io/docs/getting-started/errors#a-nameprocessingerrorsaprocessing-errors) in the documentation.  However, in the sample above, where an error is added directly to the `Errors` collection, there should be no prefix added.  Below is a sample test which passes fine on 5.3.3:

```cs
using GraphQL.Types;

namespace GraphQL.Tests.Bugs;

public class Bug3327
{
    [Fact]
    public async Task NoExcessTextOnError()
    {
        var schema = new MySchema();
        var resultJson = await schema.ExecuteAsync(o => o.Query = "{test}").ConfigureAwait(false);
        resultJson.ShouldBeCrossPlatJson("{\"errors\":[{\"message\":\"This is a test\"}],\"data\":{\"test\":null}}");
    }

    private class MySchema : Schema
    {
        public MySchema()
        {
            Query = new MyQuery();
        }
    }

    private class MyQuery : ObjectGraphType
    {
        public MyQuery()
        {
            Field<StringGraphType>("test", resolve: context =>
                {
                    context.Errors.Add(new ExecutionError("This is a test"));
                    return null;
                });
        }
    }
}
```

**Comment by Shane32** [comment]
Note that the test above uses the default configuration of the serializer, error info provider, and so on.  Written in "long form", the test looks like this:

```cs
    [Fact]
    public async Task NoExcessTextOnError()
    {
        var services = new ServiceCollection();
        services.AddGraphQL(b => b
            .AddSchema<MySchema>()
            .AddSystemTextJson());
        var provider = services.BuildServiceProvider();

        var executer = provider.GetRequiredService<IDocumentExecuter<ISchema>>();
        var result = await executer.ExecuteAsync(new ExecutionOptions
        {
            Query = "{test}",
            RequestServices = provider,
        }).ConfigureAwait(false);

        var serializer = provider.GetRequiredService<IGraphQLTextSerializer>();
        var resultJson = serializer.Serialize(result);

        resultJson.ShouldBeCrossPlatJson("{\"errors\":[{\"message\":\"This is a test\"}],\"data\":{\"test\":null}}");
    }
```

**Comment by Shane32** [comment]
If you have `ExposeExceptionStackTrace` set to `true`, as shown below, however, you will see the message in the form you described.

```cs
        services.AddGraphQL(b => b
            .AddSchema<MySchema>()
            .AddSystemTextJson()
            .AddErrorInfoProvider(o => o.ExposeExceptionStackTrace = true));
```

This is because `ExposeExceptionStackTrace` is used in this manner within `ErrorInfoProvider`:

```cs
            return new ErrorInfo
            {
                Message = _options.ExposeExceptionStackTrace ? executionError.ToString() : executionError.Message,
                Extensions = extensions,
            };
```

I suggest creating an `IErrorInfoProvider` implementation that derives from `ErrorInfoProvider`.  Override `GetInfo` and call the base method first.  Then set the `ErrorInfo.Message` property as desired before returning the instance.

**Comment by Shane32** [comment]
Note that in version 7 there is now a `ExposeExceptionDetails` property which may be what you need.  So if you were to use this in v7:

```cs
.AddErrorInfoProvider(o => o.ExposeExceptionDetails = true)
```

You would get this reply:

```json
{
  "errors": [
    {
      "message": "This is a test",
      "extensions": {
        "details": "GraphQL.ExecutionError: This is a test"
      }
    }
  ],
  "data": {
    "test": null
  }
}
```

Stack trace data, if present, would be present inside that details property, as it simply uses `exception.ToString()` similar to previously.  It just isn't in the message property.

**Comment by mariusbancila** [comment]
Yes, I do have `ExposeExceptionStackTrace` set to `true`. I am planning to upgrade to `v7` soon so then maybe I won't have to do anything. What you're showing in the last message looks what I want.

**Comment by sungam3r** [comment]
@mariusbancila I wanted to solve the same issue as you in #3276 - to see just clean message without anything else.

**Comment by sungam3r** [comment]
Fixed in #3276
